### PR TITLE
feat: Eliminate Filter Rule

### PR DESCRIPTION
**Pull request by jurplel**
TODO: Write documentation & write tests

Depends on #52

### CODE REVIEW

**Comment by yliang412** [comment]
I think this is eliminate filter rule right?

**Comment by jurplel** [comment]
Yes, thanks for the correction

File: optd-core/src/cascades/memo.rs [hunk 1]
```diff
@@ -198,6 +222,12 @@ impl<T: RelNodeTyp> Memo<T> {
         props
     }
 
+    pub fn clear_exprs_in_group(&self, group_id: ReducedGroupId) {
+        let group = self.groups.get(&group_id).expect("group not found");
```
Comment (skyzh) [inline optd-core/src/cascades/memo.rs]: this does not clear anything 🤣 either directly remove the group id from the hash map, or get a mutable reference to the entry

Comment (jurplel) [reply optd-core/src/cascades/memo.rs]: Whoops, that was extremely silly of me. Fixed!

**Review by yliang412 — APPROVED** [review]
LGTM! Thanks for the code comments!

**Pull request merged**